zerver/lib/narrow.py

```diff
@@ -597,6 +598,24 @@ def by_dm(
         )
         return query.where(maybe_negate(cond))
 
+    def _get_huddle_recipients(self, other_user: UserProfile) -> Set[int]:
+        self_recipient_ids = [
+            recipient_tuple["recipient_id"]
+            for recipient_tuple in Subscription.objects.filter(
+                user_profile=self.user_profile,
+                recipient__type=Recipient.HUDDLE,
+            ).values("recipient_id")
+        ]
+        narrow_recipient_ids = [
+            recipient_tuple["recipient_id"]
+            for recipient_tuple in Subscription.objects.filter(
+                user_profile=other_user,
+                recipient__type=Recipient.HUDDLE,
+            ).values("recipient_id")
+        ]
+
+        return set(self_recipient_ids) & set(narrow_recipient_ids)
```

Comment:
This should probably be left as a follow-up since it's orthogonal to the other changes here and will require fiddling with the query planner to make sure we're doing it right, but I'm almost certain the database queries here can be optimized by avoiding the list comprehensions that convert these into lists and letting the database layer do the final filtering...

Reply:
Yeah, these changes are big and complex enough that I feel like this as a separate follow-up task/issue makes sense.